webmention.php

```diff
@@ -70,6 +70,39 @@ public static function query_var($vars) {
   }
 
   /**
+   * generate a valid expire code. 
+   * Three possible values are valid at any one time, ticks: 0, 1, or 2
+   * 
+   * @return array
+   */
+  public static function expire_code( $tick = 0 ) {
+    $action = 'web mention endpoint';
+    $time_format = 'Y-m-d a';
+    $time_block = 12 * HOUR_IN_SECONDS;
+    $tick = abs( intval( $tick ) );
+    if ( 3 < $tick ) {
+        // something wrong, tick too high/
+        // use default
+        $tick = 0;
+    }
+    $expire_code = date( $time_format, time() - ( $tick * $time_block ) );
```

Comment:
Any reason to not use `wp_nonce_tick()` instead of calculating the tick from scratch? Not sure what's the best – but perhaps add a comment with a reason for why you chose what you did would be good?